### ISSUE TITLE
A Classes Scope with "ALL" should not require library access

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/cdisc-rules-engine#475]
Bug is a side-effect of: https://github.com/cdisc-org/cdisc-rules-engine/pull/458

Non-general observation classes require library access to determine if a domain is within a class, but if the class scope is `ALL`, this should not require library access.

For example, a rule with the following scope currently requests library information:

```yaml
Scope:
  Classes:
    Include:
      - ALL
```

**Comment by nhaydel** [comment]
I will address this today since it is an issue with my PR

**Comment by RamilCDISC** [comment]
The QA validation indicates that the issue is clear, and the acceptance criteria (AC) have been completely implemented. The engine no longer needs library access when the rule scope includes all. This conclusion is based on successful validation using various datasets and rule files. For reference, I've attached a screenshot from one of the test runs

<img width="1502" alt="475" src="https://github.com/cdisc-org/cdisc-rules-engine/assets/113539111/5031452d-d422-4fb9-b757-3969acda21e3">

**Comment by RamilCDISC** [comment]
### VALIDATION SUMMARY
### Validation Methodology
The validation process incorporated rigorous testing methodologies to ensure the accuracy and reliability of the system. This included:

**Unit Testing:**
Unit Testing was done to verify the implementation of the Acceptance criteria. Unit tests were designed to test the made changes using edge cases. All those tests were then run using pytest to verify compliance with acceptance criteria.

**Rules Engine:** 
The rules engine test (in core.py) functionality was utilized to test the proper implementation of the AC

**Code Review:**
The code changes in the code files were reviewed and were found to be in accordance with the Acceptance criteria. Also, the code was reviewed for following: 

- Code Quality Evaluation
- Logic and Functionality Review
- Error Handling and Exception Handling Assessment
- Security Considerations

**Regression Testing:**
Regression tests were run to validate that the changes did not affect any functionality of the overall system.

### Test Execution:
The tests were executed in the following order:

**Code Review:** The code for all the changes files was reviewed first to check the proper implementation of the Acceptance criteria.

**Unit Tests:** The unit tests were run using the following command in the terminal from the root of the project
python -m pytests path/to/the/test/files

**Rules Engine:** The test validation were run  with a rule where scope for class is ALL. The api key was removed from the environment to validate that library access is not required for such rules.

**Regression Tests:** The regression tests were then run to confirm that the changes in engine did not reak any other part of the implementation

### Results
After successful running of unit, Rule Engine test, regression tests and careful validation, the issue was found to be good to close as the Acceptance criteria was fulfilled and all validations passed.

### Deviations and Non-conformances:
No deviations or non-conformances were observed.

### Summary of Findings:
**Test Results:** All tests conducted during the validation process passed successfully, indicating that the system functions as intended and meets the specified requirements. The tests covered various scenarios and edge cases, providing confidence in the system's reliability and correctness.
**Unit Regression:** The unit regression tests were executed, and no regressions or issues were identified. This indicates that the changes made during the development or modification of the system did not introduce any unintended side effects or break existing functionality.
**Code Reviews:** The code reviews conducted by both development and QA teams were completed, and no major issues or concerns were identified. The code adheres to coding standards and best practices, demonstrating good code quality, readability, and maintainability.
